### PR TITLE
Refactor improve structure parsing

### PR DESCRIPTION
**Pull request by LBeernaertProton**

### CODE REVIEW

File: imap/structure.go [hunk 1]
```diff
@@ -3,130 +3,131 @@ package imap
...
-	if ext {
-		fields.
-			add(mimeParams).
-			add(getDispInfo(header)).
-			add(header.Get("Content-Language")).
-			add(header.Get("Content-Location"))
-	}
+	extWriter := writer.toSingleWriterFrom2nd()
+	fields.addMap(extWriter, mimeParams)
+	getDispInfo(fields, extWriter, header)
```
Comment (jameshoulahan) [inline imap/structure.go:54]: i'd call this `setDispInfo` or `addDispInfo`

File: imap/structure.go [hunk 2]
```diff
@@ -3,130 +3,131 @@ package imap
...
+func singlePartStructure(section *rfc822.Section, c *paramList, writer *dualParListWriter) error {
 	header, err := section.ParseHeader()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	var fields parList
+	//fields := c.newChildList()
+	//defer fields.finish()
```
Comment (jameshoulahan) [inline imap/structure.go:68]: leftover comments?

File: imap/structure.go (hunk 1 above)
Comment (LBeernaertProton) [reply imap/structure.go]: Done

File: imap/structure.go (hunk 2 above)
Comment (LBeernaertProton) [reply imap/structure.go]: Good catch, done

**Review by jameshoulahan — DISMISSED** [review]

**Review by jameshoulahan — APPROVED** [review]

**Pull request merged**